### PR TITLE
TestSpan needs to implement the AddLink functionality

### PR DESCRIPTION
**Pull request by RickWinter**
Added missing AddLink method

### CODE REVIEW

**Review by jhendrixMSFT — APPROVED** [review]

**Comment by RickWinter** [comment]
@benbp Before we enable the pipeline please see @jhendrixMSFT email on the topic

**Review by jhendrixMSFT — APPROVED** [review]

File: sdk/tracing/azotel/examples_test.go [hunk 1]
```diff
@@ -15,28 +15,31 @@ import (
...
-	// then uses the azotel adapter to conenct it to an Azure SDK client.
+	// via OTLP then uses the azotel adapter to connect it to an Azure SDK client.
 
-	// create the Jaeger exporter
-	exp, err := jaeger.New(jaeger.WithCollectorEndpoint())
+	ctx := context.Background()
+
+	// create the OTLP exporter, using the local Jaeger OTLP endpoint
+	// NOTE: this assumes you have a local instance of Jaeger running on the same machine
+	exp, err := otlptracehttp.New(ctx, otlptracehttp.WithEndpointURL("http://localhost:4318/v1/traces"))
```
[the remark this answers is not in the document]
Comment (jhendrixMSFT) [reply sdk/tracing/azotel/examples_test.go:32]: This is the documented default endpoint.

https://www.jaegertracing.io/docs/1.57/apis/#opentelemetry-protocol-stable

**Pull request merged**